packages/expect/src/matchers.ts

```diff
@@ -474,6 +474,35 @@ const matchers: MatchersObject = {
     }
 
     if (typeof received === 'string') {
+      const wrongTypeErrorMessage = `${EXPECTED_COLOR(
+        'expected',
+      )} value must be a string if ${RECEIVED_COLOR(
+        'received',
+      )} value is a string`;
+
+      if (expected == null) {
```

Comment:
My understanding of the PR is that it should prevent users from passing wrongly typed values for `toContain`. With current PR it seems that expectation like `expect("0").toContain(0);` will no longer pass while it did in version 26. (Clearly a cool change IMO)

What about those ones?
- `expect("false").toContain(false);`
- `expect("0").toContain(0n);`
- `expect("1977-04-22T06:00:00.000Z").toContain(new Date("1977-04-22T01:00:00-05:00"));`

The check for `typeof expected` only seems to handle the case of a number. But there are many other cases that may be problematic.

Reply:
they should all fail IMO. If they don't we should fix it. If they _do_ we should add tests

Reply:
I have not tested it yet to be honest. I was just surprised about the typeof check for numbers

Reply:
I confirm the fact that `expect("1").toContain(1n)` and `expect("false").toContain(false)` are passing even with the fix. I am opening a PR for that.

Reply:
PR opened: https://github.com/facebook/jest/pull/10929